processor/k8sprocessor/kube/client.go

```diff
@@ -279,15 +267,30 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
 	} else {
 		newPod.Attributes = c.extractPodAttributes(pod)
 	}
-	c.Pods[pod.Status.PodIP] = newPod
+	if pod.Status.PodIP != "" {
+		// compare initial scheduled timestamp for existing pod and new pod with same IP
+		// and only replace old pod if scheduled time of new pod is newer? This should fix
+		// the case where scheduler has assigned the same IP to a new pod but update event for
+		// the old pod came in later
+		func() {
+			if p, ok := c.Pods[pod.Status.PodIP]; ok {
+				if p.StartTime != nil && pod.Status.StartTime.Before(p.StartTime) {
```

Comment:
I think the same applies for PodUID, the flow needs to be modified a bit. BTW, I presume `pod.UID` is never `nil` nor empty?

Reply:
I'm not sure if this should be changed. There is no possibility to get the same UID for different pods

Reply:
Good point, perhaps worth including as a comment? :)